x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/transform/action/GetTransformNodeStatsAction.java

```diff
@@ -0,0 +1,143 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.xpack.core.transform.action;
+
+import org.elasticsearch.action.ActionType;
+import org.elasticsearch.action.FailedNodeException;
+import org.elasticsearch.action.support.TransportAction;
+import org.elasticsearch.action.support.nodes.BaseNodeResponse;
+import org.elasticsearch.action.support.nodes.BaseNodesRequest;
+import org.elasticsearch.action.support.nodes.BaseNodesResponse;
+import org.elasticsearch.cluster.ClusterName;
+import org.elasticsearch.cluster.node.DiscoveryNode;
+import org.elasticsearch.common.Strings;
+import org.elasticsearch.common.io.stream.StreamInput;
+import org.elasticsearch.common.io.stream.StreamOutput;
+import org.elasticsearch.rest.RestStatus;
+import org.elasticsearch.transport.TransportRequest;
+import org.elasticsearch.xcontent.ToXContent;
+import org.elasticsearch.xcontent.ToXContentObject;
+import org.elasticsearch.xcontent.XContentBuilder;
+
+import java.io.IOException;
+import java.util.List;
+
+public class GetTransformNodeStatsAction extends ActionType<GetTransformNodeStatsAction.NodesStatsResponse> {
+
+    public static final GetTransformNodeStatsAction INSTANCE = new GetTransformNodeStatsAction();
+    public static final String NAME = "cluster:admin/transform/node_stats";
+
+    private static final String TOTAL_FIELD_NAME = "total";
+    private static final String REGISTERED_TRANSFORM_COUNT_FIELD_NAME = "registered_transform_count";
+
+    private GetTransformNodeStatsAction() {
+        super(NAME);
+    }
+
+    public static class NodesStatsRequest extends BaseNodesRequest<NodesStatsRequest> {
+
+        public NodesStatsRequest() {
+            super(Strings.EMPTY_ARRAY);
+        }
+
+        @Override
+        public void writeTo(StreamOutput out) throws IOException {
+            TransportAction.localOnly();
+        }
+    }
+
+    public static class NodesStatsResponse extends BaseNodesResponse<NodeStatsResponse> implements ToXContentObject {
+
+        public int getTotalRegisteredTransformCount() {
+            int totalRegisteredTransformCount = 0;
+            for (var nodeResponse : getNodes()) {
```

Comment:
I verified locally and it worked out-of-the-box.
How I did that?
1. Build 2 elasticsearch distributions, one from `main`, the other one from local branch
2. Start `bin/elasticsearch` on 2 different ports
3. These 2 nodes will form a cluster
4. Issue 2 calls to `_transform/_node_stats` API:
a) To the "old" node:
`curl --insecure -u "elastic:yao4++Fl1SIff4v4=ftZ" -X GET https://localhost:9200/_transform/_node_stats | jq`
yields
```
{
  "error": {
    "root_cause": [
      {
        "type": "resource_not_found_exception",
        "reason": "Transform with id [_node_stats] could not be found"
      }
    ],
    "type": "resource_not_found_exception",
    "reason": "Transform with id [_node_stats] could not be found"
  },
  "status": 404
}
```
which is ok since the "old" node doesn't know about this new API

b) to the "new" node:
`curl --insecure -u "elastic:yao4++Fl1SIff4v4=ftZ" -X GET https://localhost:9201/_transform/_node_stats | jq`
yields
```
{
  "q4vZOUrxRuq9E531PeTJIw": {
    "registered_transform_count": 5
  },
  "total": {
    "registered_transform_count": 5
  }
}
```

5. Notice that the response only contains the "new" node and omits the "old" node.